src/wasi/preview_1.rs

```diff
@@ -0,0 +1,114 @@
+#[link(wasm_import_module = "wasi_snapshot_preview1")]
+unsafe extern "C" {
+    #[link_name = "path_open"]
+    pub unsafe fn path_open(
+        dirfd: i32,
+        dirflags: i32,
+        path: i32,
+        path_len: i32,
+        oflags: i32,
+        fs_rights_base: i64,
+        _fs_rights_inheriting: i64,
+        fdflags: i32,
+        fd_ptr: i32,
+    ) -> i32;
+
+    #[link_name = "path_create_directory"]
+    pub unsafe fn path_create_directory(dirfd: i32, path_ptr: i32, path_len: i32) -> i32;
+
+    #[link_name = "path_remove_directory"]
+    pub unsafe fn path_remove_directory(dirfd: i32, path_ptr: i32, path_len: i32) -> i32;
+
+    #[link_name = "path_unlink_file"]
+    pub unsafe fn path_unlink_file(dirfd: i32, path_ptr: i32, path_len: i32) -> i32;
+
+    #[link_name = "path_symlink"]
+    pub unsafe fn path_symlink(
+        old_path_ptr: i32,
+        old_path_len: i32,
+        dirfd: i32,
+        new_path_ptr: i32,
+        new_path_len: i32,
+    ) -> i32;
+
+    #[link_name = "fd_close"]
+    pub unsafe fn fd_close(fd: i32) -> i32;
+
+    #[link_name = "fd_sync"]
+    pub unsafe fn fd_sync(fd: i32) -> i32;
+
+    #[link_name = "fd_datasync"]
+    pub unsafe fn fd_datasync(fd: i32) -> i32;
+
+    #[link_name = "fd_prestat_get"]
+    pub unsafe fn fd_prestat_get(fd: i32, path_len: i32) -> i32;
```

Comment:
param needs alignment with wasi spec:
The second parameter is a pointer to a struct, not a length
```suggestion
    pub unsafe fn fd_prestat_get(fd: i32, prestat_ptr: i32) -> i32;
```
source: https://wasix.org/docs/api-reference/wasi/fd_prestat_get#syntax